dlc-wallet/src/main/scala/org/bitcoins/dlc/wallet/DLCWallet.scala

```diff
@@ -251,22 +251,23 @@ abstract class DLCWallet
     */
   override def cancelDLC(dlcId: Sha256Digest): Future[Unit] = {
     for {
-      inputs <- dlcInputsDAO.findByDLCId(dlcId)
-      dbs <- spendingInfoDAO.findByOutPoints(inputs.map(_.outPoint))
-      // allow this to fail in the case they have already been unreserved
-      _ <- unmarkUTXOsAsReserved(dbs).recover { case _: Throwable => () }
-
       dlcOpt <- dlcDAO.read(dlcId)
-      _ = dlcOpt match {
+      dlcDb = dlcOpt match {
         case Some(db) =>
           require(
             db.state == DLCState.Offered || db.state == DLCState.Accepted || db.state == DLCState.Signed,
             "Cannot cancel a DLC after it has been signed")
+          db
         case None =>
           throw new IllegalArgumentException(
             s"No DLC Found with dlc id ${dlcId.hex}")
       }
 
+      inputs <- dlcInputsDAO.findByDLCId(dlcId, dlcDb.isInitiator)
+      dbs <- spendingInfoDAO.findByOutPoints(inputs.map(_.outPoint))
+      // allow this to fail in the case they have already been unreserved
+      _ <- unmarkUTXOsAsReserved(dbs).recover { case _: Throwable => () }
```

Comment:
it would be nice if we had a specific error here. `Throwable` can represent all sorts of things

Reply:
Ugh, i guess this is from before. We should probably offer an `Option` here on the `unmarkUTXOsAsReserved` method or something like that.

Reply:
Yeah, imo we should always recover from this, we want to cancel the DLC still, we can always unreserve later